drivers/dma/dma_stm32.h

```diff
@@ -26,7 +26,7 @@ struct dma_stm32_stream {
 	uint32_t src_size;
 	uint32_t dst_size;
 	void *user_data; /* holds the client data */
-	dma_callback_t dma_callback;
+	dma_callback_t dma_callback; /* with status = 1 for half-complete transfer */
```

Comment:
Status values are defined by DMA API as [[`include/zephyr/drivers/dma.h`](https://github.com/zephyrproject-rtos/zephyr/blame/dae6d2f84b47cd6c295296dde7dd7d97c40d652d/include/zephyr/drivers/dma.h#LL141)](https://github.com/zephyrproject-rtos/zephyr/blob/dae6d2f84b47cd6c295296dde7dd7d97c40d652d/include/zephyr/drivers/dma.h#L141).

Adding more values would require api change, which is possible but requires discussion and approval from @teburd.
What about using `dma_get_status` existing API ?

Reply:
Indeed this was my main concern regarding this PR and I don't really like having to use a custom status as well.

As written in the PR description, `dma_get_status` isn't used because the indexes in `struct dma_status` can be modified by the time `uart_stm32_dma_rx_flush` is executed even though the `uart_stm32_dma_rx_flush` is called because the buffer is half or fully filled and the RX buffer should be handled accordingly.

Reply:
> As written in the PR description,

Right, missed it! txs

Reply:
What's the status on this? 🤔 
Should we make any positive returned value a custom status that can be used differently depending on the platform?
Microcontrollers have different DMA interrupts so I don't think we want to force the callback to return the same status codes on every MCU.
Should we create an issue for this so that we can discuss it?
cc @teburd

Reply:
> Should we create an issue for this so that we can discuss it?

This would be the most efficient.

Reply:
That sounds like a good plan

Reply:
See #52417 

I actually found that `dma_dw_common.h` stores 2 callbacks in the driver data which might be a good alternative.